code/game/machinery/robotic_arm.dm

```diff
@@ -0,0 +1,232 @@
+	. = ..()
+	wires = new /datum/wires/robotic_arm(src)
+	dummy = new /mob/living/simple_animal/robotic_arm_dummy(src)
+	dummy.arm = src
+	source_update()
+
+/obj/machinery/robotic_arm/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS, null, CALLBACK(src, .proc/can_be_rotated))
+
+/obj/machinery/robotic_arm/proc/can_be_rotated(mob/user,rotation_type)
+	if (anchored)
+		to_chat(user, "<span class='warning'>It is fastened to the floor!</span>")
+		return FALSE
+	return TRUE
+
+/obj/machinery/robotic_arm/Moved() //In case we get moved while still anchored (admemes, shuttle, whatever)
+	. = ..()
+	if(!anchored)
+		return
+	set_turfs()
+
+/obj/machinery/robotic_arm/attackby(obj/item/I, mob/user, params)
+	if(is_wire_tool(I) && panel_open)
+		wires.interact(user)
+		return
+	return ..()
+
+/obj/machinery/robotic_arm/wrench_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	if(!I.use_tool(src, user, 5, volume=50))
+		return 
+	anchored = !anchored
+	if(!anchored)
+		STOP_PROCESSING(SSmachines, src)
+	else
+		set_turfs()
+		START_PROCESSING(SSmachines, src)
+	user.visible_message("<span class='notice'>[user.name] [anchored?"unbolts":"bolts down"] the [src].</span>", \
+				"<span class='notice'>You [anchored?"unbolt":"bolt down"] the [src].</span>")
+	return TRUE
+
+/obj/machinery/robotic_arm/screwdriver_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	default_deconstruction_screwdriver(user, "robot_arm", "robot_arm", I)
+	return TRUE
+
+/obj/machinery/robotic_arm/crowbar_act(mob/living/user, obj/item/I)
+	if(!panel_open)
+		return FALSE
+	if(I.use_tool(src, user, 5, volume=50))
+		if(r_arm)
+			new /obj/item/bodypart/r_arm/robot(get_turf(src))
+		else
+			new /obj/item/bodypart/l_arm/robot(get_turf(src))
+		new /obj/item/robotic_arm_base(get_turf(src))
+		user.visible_message("<span class='notice'>[user.name] prys off the robotic arm from its base.</span>", \
+				"<span class='notice'>You pry off the robotic arm from its base</span>")
+		qdel(dummy) //prevent robo-gibs
+		qdel(src)
+		return TRUE
+
+/obj/machinery/robotic_arm/Destroy()
+	target = null
+	source = null
+	dest = null
+	if(held)
+		held.forceMove(get_turf(src))
+	held = null
+	if(dummy && !QDELETED(dummy))
+		dummy.death(FALSE)
+	. = ..()
+
+/obj/machinery/robotic_arm/proc/set_turfs()
+	if(source)
+		UnregisterSignal(source, COMSIG_TURF_CONTENTS_CHANGE)
+	if(dest)
+		UnregisterSignal(dest, COMSIG_TURF_CONTENTS_CHANGE)
+	source = get_step(src, turn(dir,-180))
+	RegisterSignal(source, COMSIG_TURF_CONTENTS_CHANGE, .proc/source_update)
+	dest = get_step(src, dir)
+
+/obj/machinery/robotic_arm/process()
+	..()
+	if(!anchored || moving)
+		return
+	if(!anchored  || wires.is_cut(WIRE_SENSOR) || wires.is_cut(WIRE_MOTOR1) || !locate(/obj/item) in source) //If we're unanchored, our detection wire is cut, or the arm rotate wire is cut
```

Comment:
Shouldn't `!locate(/obj/item) in source` be replaced with the `!grab()` seen below, and the one below being removed?

Reply:
The `grab()` proc returns true even if the claw alread had something (due to the motor2 wire cut meaning the claw can't drop), which was resulting in a claw that was holding an item swinging back and forth even if the source tile was empty. Hence why the check was copied here.